gtsfm/frontend/global_descriptor/netvlad_global_descriptor.py

```diff
@@ -0,0 +1,45 @@
+"""Wrapper around the NetVLAD global image descriptor.
+
+Based on Arandjelovic16cvpr:
+"NetVLAD: CNN architecture for weakly supervised place recognition"
+https://arxiv.org/pdf/1511.07247.pdf
+
+NetVLAD, is a new generalized VLAD layer, inspired by the “Vector of Locally Aggregated Descriptors”
+image representation commonly used in image retrieval
+
+Whereas bag-of-visual-words aggregation keeps counts of visual words, VLAD stores the sum of residuals
+(difference vector between the descriptor and its corresponding cluster centre) for each visual word.
+
+Authors: John Lambert
+"""
+
+import numpy as np
+import torch
+from torch import nn
+
+from gtsfm.common.image import Image
+from gtsfm.frontend.global_descriptor.global_descriptor_base import GlobalDescriptorBase
+from thirdparty.hloc.netvlad import NetVLAD
+
+
+class NetVLADGlobalDescriptor(GlobalDescriptorBase):
+    def __init__(self) -> None:
+        """ """
+        pass
+
+    def describe(self, image: Image) -> np.ndarray:
+        """
+        Args:
+            image: input image.
+
+        Returns:
+            img_desc: array of shape (D,) representing global image descriptor.
+        """
+        model: nn.Module = NetVLAD()
```

Comment:
why is this not done in `__init__()` ?

Reply:
initializing in the constructor leads to OOM w/ our current Dask paradigm.

Reply:
how/why? 🤔

Reply:
I haven't dug into it too much to be honest. But I tested it both ways, and using it in the constructor was essentially unrunnable. I believe it has to do with the amount of memory being allocated and kept around per worker.

Reply:
unrunnable on the CI or locally as well? 
I find issues when running with a high max lookahead from distributed utils complaining that 20-50% of CPU is being used for garbage collection. This could be another instance where that might happen. We should not have inefficient code just to fit a 500 image dataset in 16GB RAM, which is kind of a long shot.

Reply:
unrunnable locally for me